File: server/models/message.go
package models

type PendingMessage struct {
	ID          string
	RecipientID string
	Ciphertext  []byte
	Delivered   bool
}

type SenderKeyDistribution struct {
	ID          string
	GroupID     string
	RecipientID string
	Ciphertext  []byte
	Delivered   bool
}

File: client/src/types/message.ts
export interface DirectMessage {
  recipient_id: string;
  sender_id?: string;
  content: string;
  timestamp?: number;
}

export interface GroupMessage {
  group_id: string;
  sender_id?: string;
  content: string;
  timestamp?: number;
}

export interface SenderKeyDistribution {
  group_id: string;
  content: string;
}
